broker/data_upload/_ingest_alerts.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
+"""Parse ZTF alerts and add them to the project database."""
+
+import json
+import os
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+import fastavro
+import pandavro as pdx
+
+from .._utils import setup_log
+
+if 'RTD_BUILD' not in os.environ:
+    from google.cloud import error_reporting, bigquery, storage
+
+    error_client = error_reporting.Client()
+    bq_client = bigquery.Client()
+    log = setup_log('data_upload')
+
+SCHEMA_DIR = FILE_DIR = Path(__file__).resolve().parent / 'schema'
+
+
+def _get_table_id(data_set, table):
+    """Return the ID for a BigQuery table
+
+    Args:
+        data_set (str): The name of the data set
+        table    (str): The name of the table
+
+    Returns:
+        The name of the specified table as a string
+    """
+
+    table_ref = bq_client.dataset(data_set).table(table)
+    return f'{table_ref.dataset_id}.{table_ref.table_id}'
+
+
+def stream_ingest(data, data_set, table):
+    """Stream ingest a Pandas DataFrame into a BigQuery table
+
+    If the table does not exist, create it.
+
+    Args:
+        data (DataFrame): Data to upload to table
+        data_set   (str): The name of the data set
+        table      (str): The name of the table
+    """
+
+    project_id = os.environ['BROKER_PROJ_ID']
+    table_id = _get_table_id(data_set, table)
+    data.to_gbq(
+        table_id,
+        project_id,
+        if_exists='append',
+        progress_bar=False
+    )
+
+
+def batch_ingest(data, data_set, table):
+    """Batch upload a Pandas DataFrame into a BigQuery table
+
+    Alert data WILL be temporarily written to disk. If the table does not
```

Comment:
I suppose.  Whether it actually gets written to disk is up to the kernel and file system.  But, yes, we should flag this for later review to see if creating things purely in memory to upload performs better.